src/core/federation.ts

```diff
@@ -281,6 +287,123 @@ function isANodeJSCodedError(e: Error): e is NodeJSCodedError {
   return typeof (e as NodeJSCodedError).code === "string";
 }
 
+/**
+ * A network sender is a reactor containing a portAbsentReaction.
+ */
+export class NetworkSender extends Reactor {
+  /**
+   * The last reaction of a NetworkSender reactor is 'portAbsentReactor'.
+   * @returns portAbsentReactor of this NetworkSender reactor
+   */
+  public getLastReactioOrMutation(): Reaction<Variable[]> | undefined {
+    return this._getLastReactionOrMutation();
+  }
+}
+
+/**
+ * A network receiver is a reactor handling a network input.
+ */
+export class NetworkReceiver<T> extends Reactor {
+  /*
+   * A FederatePortAction instance of this NetworkReceiver.
+   */
+  private networkInputAction: FederatePortAction<T> | undefined = undefined;
```

Comment:
I don't think you need to set it to `undefined` explicitly?
```suggestion
  private networkInputAction: FederatePortAction<T> | undefined;
```

Reply:
It depends, but it either needs to be initialised here, or in `constructor`. I suggest moving it into constructor, because `portID` is initialised there as well.

Reply:
As @axmmisaka said, I set it to `undefined` to initialize it. I'll move it into constructor.